ArgusProbe_Beamlogic.py

```diff
@@ -227,12 +227,113 @@ def _get_ntp_timestamp():
+    """
+    Thread which attaches to the serial and put frames into queue.
+    """
+
+    def __init__(self, txMqttThread, serialport):
+
+        # store params
+        self.txMqttThread             = txMqttThread
+        self.serialport               = serialport
+        self.baudrate                 = 115200          #Should i put it in input params?
+
+        # local variables
+        self.serialHandler           = None
+        self.rxBuffer                = []               #keep the frame
+        self.goOn                    = True
+        self.pleaseConnect           = True
+        self.dataLock                = threading.RLock()
+
+
+        # initialize thread
+        super(Serial_RxSnifferThread, self).__init__()
+        self.name                 = 'Serial_RxSnifferThread@{0}'.format(self.serialport)
+        self.start()
+
+    def run(self):
+
+        time.sleep(1)  # let the banners print
+        while self.goOn:
+            try:
+                with self.dataLock:
+                    pleaseConnect = self.pleaseConnect
+
+                if pleaseConnect:
+                    # open serial port
+                    self.serialHandler = serial.Serial(self.serialport, baudrate=self.baudrate)
+
+                    # read byte
+                    while True:
+                        waitingbytes   = self.serialHandler.inWaiting()
+                        if waitingbytes != 0:
+                            c = self.serialHandler.read(waitingbytes)
+                            self._newByte(c)
+                            print (c)
+                            time.sleep(0.2)
+
+            except serial.SerialException:
+                # mote disconnected, or pyserialHandler closed
+                # destroy pyserial instance
+                print "WARNING: Could not read from serial at \"{0}\".".format(
+                        self.serialport)
+                print "Is device connected?"
+                self.goOn            = False
+                self.serialHandler   = None
+                time.sleep(1)
+
+
+            except Exception as err:
+                logCrash(self.name, err)
+
+
+    #======================== public ==========================================
+
+    def connectSerialPort(self):
+        with self.dataLock:
+            self.pleaseConnect = True
+
+    def disconnectSerialPort(self):
+        with self.dataLock:
+            self.pleaseConnect = False
+        try:
+            self.serialHandler.close()
+        except:
+            pass
+
+    def close(self):
+        self.goOn            = False
+    #======================== public ==========================================
+
+    #======================== private =========================================
+
+    def _newByte(self, b):
+        """
+        Just received a byte from the sniffer
+        """
+        with self.dataLock:
+            self.rxBuffer      += [b]
+
+        self._newFrame(self.rxBuffer)
```

Comment:
I understand that lines #319 and #329 are temporary code that publishes each received byte over MQTT. If so, this code should not really belong within the `_newByte` method but rather within the thread that calls it (`run()`).